docs: Importing JSON files correctly outside GraphQL (#17061)
gatsbyjs/gatsby: docs/docs/static-folder.md

  [`pace.js`](http://github.hubspot.com/pace/docs/welcome/) outside of the
  bundled code.
- Some libraries may be incompatible with Webpack and you have no other option but to include it as a `<script>` tag.

> - You need to import JSON file that doesn't have a consistent schema, like [TopoJSON files](https://en.wikipedia.org/wiki/GeoJSON#TopoJSON), which GraphQL can't handle correctly. Note that importing JSON files directly inside a page, a template, or a component using `import` syntax results in adding that file to the app bundle and increasing the size of all site's pages. Instead, it's better to place your JSON file inside the `static` folder and then fetch that file using something like [axios](https://github.com/axios/axios/) within `componentDidMount` lifecycle or `useEffect` hook.

Reviewer: WDYT of? Mainly your words but mixed in the dynamic import part.
```suggestion
- You need to import JSON file that doesn't have a consistent schema, like [TopoJSON files](https://en.wikipedia.org/wiki/GeoJSON#TopoJSON), which is difficult to handle with GraphQL. Note that importing JSON files directly inside a page, a template, or a component using `import` syntax results in adding that file to the app bundle and increasing the size of all site's pages. Instead, it's better to place your JSON file inside the `static` folder and use the dynamic import syntax (`import('/static/myjson.json')`) within the `componentDidMount` lifecycle or the `useEffect` hook
```

Reviewer 2: I think @wardpeet advice here makes a lot of sense. Given that it changes actual advice from using `static` directory to using webpack's dynamic import - I think it should also be moved to different documentation page - for example https://www.gatsbyjs.org/docs/importing-assets-into-files/#importing-assets-with-webpack

Author: Hello again, I'm sorry to be late on this. 
@pieh I think it's important to add the same advice to gatsby-transformer-json`s README file too, as the plugin couldn't handle files which have a random schema. If it sounds OK I'm ready to send a small commit to add the same mention there.

Reviewer 2: Yeah, sounds reasonable to mention "gotcha"s in json transformer docs. Just one note - json transformer isn't limited to local files (it can work on content pulled from CMSes), so the advice for objects without consistent schema will not apply to all potential scenarios

Reviewer 3: in the import from webpack there is no mention of `json` maybe it should also added there?

Reviewer 3: ```
To reduce the number of requests to the server, importing images that are less than 10,000 bytes returns a data URI instead of a path. This applies to the following file extensions: svg, jpg, jpeg, png, gif, mp4, webm, wav, mp3, m4a, aac, and oga.
```

oh i think i misunderstood the second sentence
